lib/rubocop/cop/style/indent_heredoc.rb

```diff
@@ -87,14 +120,27 @@ def correct_by_library(node)
           end
         end
 
+        def check_style!
+          case style
+          when nil
+            raise Warning, "Auto Correction does not work for #{cop_name}. " \
```

Comment:
> Auto-correction

👍 

> why should we check whether enforce style is set or not? 

This `nil` checking does not check set or not.
The `style` method returns `nil` if auto detection can't decide style. So, the `nil` checking checks the detection. 

I think this code is a bit confusing. I'll add a comment.


> And why should we do it in the auto-correction?


I think this check should be delayed until auto-correction. Because user doesn't necessarily use auto-correction. In that case user doesn't need the warnings.